src/server/scripts/Spells/spell_shaman.cpp

```diff
@@ -1043,6 +1046,38 @@ class spell_sha_path_of_flames_spread : public SpellScript
     }
 };
 
+// 2645 - Ghost Wolf
+class spell_sha_ghost_wolf : public AuraScript
+{
+    PrepareAuraScript(spell_sha_ghost_wolf);
+
+    bool Validate(SpellInfo const* /*spellInfo*/) override
+    {
+        return ValidateSpellInfo({ SPELL_SHAMAN_SPIRIT_WOLF_TALENT, SPELL_SHAMAN_SPIRIT_WOLF_PERIODIC, SPELL_SHAMAN_SPIRIT_WOLF_AURA });
+    }
+
+    void OnApply(AuraEffect const* /*aurEff*/, AuraEffectHandleModes /*mode*/)
+    {
+        if (GetTarget()->HasAura(SPELL_SHAMAN_SPIRIT_WOLF_TALENT))
+            GetTarget()->CastSpell(GetTarget(), SPELL_SHAMAN_SPIRIT_WOLF_PERIODIC, true);
+    }
+
+    void OnRemove(AuraEffect const* /*aurEff*/, AuraEffectHandleModes /*mode*/)
+    {
+        if (GetTarget()->HasAura(SPELL_SHAMAN_SPIRIT_WOLF_TALENT))
+        {
+            GetTarget()->RemoveAurasDueToSpell(SPELL_SHAMAN_SPIRIT_WOLF_PERIODIC);
+            GetTarget()->RemoveAurasDueToSpell(SPELL_SHAMAN_SPIRIT_WOLF_AURA);
+        }
+    }
+
+    void Register() override
+    {
+        OnEffectApply += AuraEffectApplyFn(spell_sha_ghost_wolf::OnApply, EFFECT_0, SPELL_AURA_MOD_SHAPESHIFT, AURA_EFFECT_HANDLE_REAL);
```

Comment:
When you cast spells or remove auras inside apply/remove hooks, you must use AfterEffectApply and AfterEffectRemove, not `On*`

Reply:
oh ok thanks :D